pallet/vortex-distribution/src/lib.rs

```diff
@@ -696,51 +1013,157 @@ pub mod pallet {
 				T::MultiCurrency::balance(T::NativeAssetId::get(), &root_vault_account);
+			let root_vault_root_value: BalanceOf<T> = root_vault_root_token_balance * root_price;
 			Self::safe_transfer(
 				T::NativeAssetId::get(),
 				&root_vault_account,
-				&vault_account,
+				&vtx_vault_account,
 				root_vault_root_token_balance,
 				false,
 			)?;
 
-			VtxDistStatuses::<T>::mutate(id, |status| {
-				*status = VtxDistStatus::Triggered;
-			});
-			Self::deposit_event(Event::TriggerVtxDistribution { id });
+			// fetch calculated vortex price
+			let vortex_price = VtxPrice::<T>::get(id);
+			ensure!(vortex_price > Zero::zero(), Error::<T>::VortexPriceIsZero);
+
+			//calculate total rewards
+			let total_vortex_network_reward = fee_vault_asset_value / vortex_price;
+			let total_vortex_bootstrap = root_vault_root_value / vortex_price;
+			let total_vortex = total_vortex_network_reward + total_vortex_bootstrap;
+
+			// store TotalVortex only if EnableManualRewardInput is false
+			// otherwise in manual mode the TotalVortex will be calculated from the input.
+			if !EnableManualRewardInput::<T>::get() {
+				TotalVortex::<T>::insert(id, total_vortex);
+			}
+
+			TotalNetworkReward::<T>::insert(id, total_vortex_network_reward);
+			TotalBootstrapReward::<T>::insert(id, total_vortex_bootstrap);
 
 			Ok(().into())
 		}
 
+		fn do_reward_calculation(id: T::VtxDistIdentifier) -> DispatchResultWithPostInfo {
+			// fetch and calculate reward pool balances
+			let total_network_reward = TotalNetworkReward::<T>::get(id);
+			let total_bootstrap_reward = TotalBootstrapReward::<T>::get(id);
+			// Ref -> https://docs.therootnetwork.com/intro/learn/tokenomics#how-are-rewards-distributed
+			let total_staker_pool =
+				total_bootstrap_reward + Perbill::from_percent(30) * total_network_reward; // bootstrap + 30% of network rewards
+			let total_workpoints_pool = Perbill::from_percent(70) * total_network_reward; // 70% of network rewards
+			let total_staker_points = TotalRewardPoints::<T>::get(id);
+			let total_work_points = TotalWorkPoints::<T>::get(id);
+
+			let mut account_ids: BoundedVec<T::AccountId, T::MaxRewards> = BoundedVec::default();
+			// Iterate RewardPoints to capture all the accounts in this reward cycle.
+			// This means all validators, nominators and stakers should be in this map.
+			// Note that all accounts, even with 0 staker rewards must be registered onchain
+			// as this is the reference to capture all accounts for this cycle.
+			for (account_id, _) in RewardPoints::<T>::iter_prefix(id) {
+				if !account_ids.contains(&account_id) {
+					account_ids.try_push(account_id).map_err(|_| Error::<T>::ExceededMaxRewards)?;
+				}
+			}
+
+			// Calculate and register each account's reward portion
+			for account_id in &account_ids {
+				let account_staker_points: BalanceOf<T> =
+					RewardPoints::<T>::get(id, account_id.clone());
+				let account_work_points: BalanceOf<T> =
+					WorkPoints::<T>::get(id, account_id.clone());
```

Comment:
No need to use iter_prefix and then iterate through the cached result. Instead could do all calculations in the iter_prefix part and fetch the balance immediately
```suggestion
			for (account_id, account_staker_points) in RewardPoints::<T>::iter_prefix(id) {
				let account_work_points: BalanceOf<T> =
					WorkPoints::<T>::get(id, account_id.clone());
```

Reply:
a66e2176e79faf6824b25b455fb13e5b6b5d646e